dataflow/src/domain/mod.rs

```diff
@@ -1974,6 +2003,18 @@ impl Domain {
                                     n.with_reader_mut(|r| {
                                         r.writer_mut().map(|wh| {
                                             for miss in &missed_on {
+                                                use core::data::SizeOf;
+
+                                                let size = wh.with_key(&miss[..])
+                                                    .try_find_and(|rs| {
+                                                        rs.iter().map(|r| r.deep_size_of()).sum()
+                                                    })
+                                                    .map(|r| r.0.unwrap_or(0))
+                                                    .unwrap_or(0);
+
+                                                wh.mem_size =
+                                                    wh.mem_size.checked_sub(size as usize).unwrap();
```

Comment:
This is the thing I mentioned when we talked. It can't be done in `mark_hole` because we do not have access to the `WriteHandle`'s `self.mem_size` there.

Reply:
Hmm... That seems fixable. Just make https://github.com/mit-pdos/distributary/blob/master/dataflow/src/backlog/mod.rs#L120 be an `&'a mut WriteHandle` instead?